### PR TITLE
New: valid-v-slot rule (fixes #802)

### LINKED ISSUE #802: Rule Proposal: vue/valid-v-slot
**Please describe what the rule should do:**

This rule reports wrong usage of `v-slot` directives.

This is `v-slot` version of existing `vue/valid-v-*` rules.

**What category should the rule belong to?**

- [ ] Enforces code style
- [X] Warns about a potential error
- [ ] Suggests an alternate way of doing something
- [ ] Other (please specify:)

**Provide 2-3 code examples that this rule should warn about:**

```vue
<template>
    <div v-slot="foo"></div><!-- error: 'v-slot' directive must be owned by a custom element, but '<div>' is not.  -->
</template>
```

```vue
<template>
    <div>
        <template v-slot:header></template><!-- error: 'v-slot' directive must be owned by a custom element, but '<div>' is not.  -->
    </div>
</template>
```

```vue
<template>
    <template v-slot:header></template><!-- error: 'v-slot' directive must be owned by a custom element, but '<template>' is not.  -->
</template>
```

```vue
<template>
    <my-component v-slot:one="foo" v-slot:two="foo"></my-component><!-- error: an element cannot have multiple 'v-slot' directives.  -->
</template>
```

```vue
<template>
    <my-component><template v-slot:one="foo" v-slot:two="foo"></template></my-component><!-- error: an element cannot have multiple 'v-slot' directives.  -->
</template>
```

```vue
<template>
    <my-component v-slot.foo="foo"></my-component><!-- error: 'v-slot' doesn't have any modifiers.  -->
</template>
```

```vue
<template>
    <my-component v-slot:header="foo"></my-component><!-- error: Named slots must be on '<template>' elements on a custom element.  -->
</template>
```

```vue
<template>
    <my-component v-slot><!-- error: Default slot should be on '<template>' elements on a custom element when other named slots exist.  -->
        <template v-slot:header></template>
    </my-component>
</template>
```

```vue
<template>
    <my-component v-slot></my-component><!-- error: Default slot on a custom element directly requires attribute value.  -->
</template>
```

Are there other cases?

**Additional context**

### PR DESCRIPTION
**Pull request by mysticatea**
Fixes #802.

### CODE REVIEW

File: docs/rules/valid-v-slot.md [hunk 1]
```diff
@@ -0,0 +1,112 @@
...
+  <my-component>
+    <template v-slot:default>
+      default
+    <template>
+    <template v-slot:one>
+      one
+    <template>
+    <template v-slot:two>
+      two
+    <template>
```
Comment (ota-meshi) [inline docs/rules/valid-v-slot.md]: These 3 template elements are not closed.

```suggestion
    </template>
```

Comment (mysticatea) [reply docs/rules/valid-v-slot.md]: Thank you!

File: docs/rules/valid-v-slot.md [hunk 2]
```diff
@@ -0,0 +1,112 @@
...
+  <my-component v-slot="data">
+    {{data}}
+  </my-component>
+  <my-component>
+    <template v-slot:default>
+      default
+    <template>
+    <template v-slot:one>
+      one
+    <template>
```
Comment (ota-meshi) [inline docs/rules/valid-v-slot.md]: I'm sorry. I overlooked it. The two above it were not closed as well.

Comment (mysticatea) [reply docs/rules/valid-v-slot.md]: Oops, thank you!

**Pull request merged**